core/storage/src/storages/beacon.rs

```diff
@@ -22,6 +32,74 @@ impl GenericStorage<SignedBeaconBlockHeader, SignedBeaconBlock> for BeaconChainS
 
 impl BeaconChainStorage {
     pub fn new(storage: Arc<KeyValueDB>) -> Self {
-        Self { generic_storage: BlockChainStorage::new(storage, ChainId::BeaconChain) }
+        Self { 
+            generic_storage: BlockChainStorage::new(storage, ChainId::BeaconChain),
+        }
+    }
+
+    pub fn get_proposal(&mut self) -> HashMap<Slot, Vec<AuthorityStake>> {
+        read_one_column(self.generic_storage.storage.as_ref(), COL_PROPOSAL)
+    }
+
+    pub fn set_proposal(&mut self, proposal: &HashMap<Slot, Vec<AuthorityStake>>) {
+        write_one_column(self.generic_storage.storage.as_ref(), COL_PROPOSAL, proposal);
+    }
+
+    pub fn get_participation(&mut self) -> HashMap<Slot, AuthorityMask> {
+        read_one_column(self.generic_storage.storage.as_ref(), COL_PARTICIPATION)
+    }
+
+    pub fn set_participation(&mut self, participation: &HashMap<Slot, AuthorityMask>) {
+        write_one_column(self.generic_storage.storage.as_ref(), COL_PARTICIPATION, participation);
+    }
+
+    pub fn get_processed_blocks(&mut self) -> HashMap<Epoch, HashSet<Slot>> {
+        read_one_column(self.generic_storage.storage.as_ref(), COL_PROCESSED_BLOCKS)
+    }
+
+    pub fn set_processed_blocks(&mut self, processed_blocks: &HashMap<Epoch, HashSet<Slot>>) {
+        write_one_column(self.generic_storage.storage.as_ref(), COL_PROCESSED_BLOCKS, processed_blocks);
     }
+
+    pub fn get_threshold(&mut self) -> HashMap<Epoch, u64> {
+        read_one_column(self.generic_storage.storage.as_ref(), COL_THRESHOLD)
+    }
+
+    pub fn set_threshold(&mut self, threshold: &HashMap<Epoch, u64>) {
+        write_one_column(self.generic_storage.storage.as_ref(), COL_THRESHOLD, threshold);
+    }
+
+    pub fn get_accepted_authorities(&mut self) -> HashMap<Slot, Vec<AuthorityStake>> {
+        read_one_column(self.generic_storage.storage.as_ref(), COL_ACCEPTED_AUTHORITY)
+    }
+
+    pub fn set_accepted_authorities(&mut self, accepted_authorities: &HashMap<Slot, Vec<AuthorityStake>>) {
+        write_one_column(self.generic_storage.storage.as_ref(), COL_ACCEPTED_AUTHORITY, accepted_authorities);
+    }
+}
+
+fn read_one_column<K, V>(storage: &KeyValueDB, col: u32) -> HashMap<K, V>
```

Comment:
This implementation is going to be very expensive. We should not be retrieving the entire column from the storage.

Reply:
We only do this when we initialize the authority struct, which does not happen very often

Reply:
This is still very inefficient. We should not be loading all of the objects in the cache upon initialization. What should be happening is whenever someone requests information on slot X, this information is retrieved from the storage, added to cache and the reference is returned.